plugins/obs-webrtc/whip-output.cpp

```diff
@@ -347,13 +349,16 @@ bool WHIPOutput::Connect()
 		return false;
 	}
 
-	if (location_header.empty()) {
+	long redirect_count = 0;
+	curl_easy_getinfo(c, CURLINFO_REDIRECT_COUNT, &redirect_count);
+
+	if (location_headers.size() < static_cast<size_t>(redirect_count) + 1) {
 		do_log(LOG_WARNING,
 		       "WHIP server did not provide a resource URL via the Location header");
```

Comment:
So I still have one question, when we make requests to a whip endpoint and they don't send us a location header does that mean that we actually cannot send delete / update requests? Or are the providers incorrectly implementing their endpoints so that the endpoint also receives those requests?

Any thoughts on if we could fail these and show an alert to put pressure on these services that have an incorrect implementation?

Reply:
Reference to [WHIP Section 4](https://datatracker.ietf.org/doc/html/draft-ietf-wish-whip-09#name-protocol-operation)

> To explicitly terminate a session, the WHIP client MUST perform an HTTP DELETE request to the resource URL returned in the Location header field of the initial HTTP POST. Upon receiving the HTTP DELETE request, the WHIP resource will be removed and the resources freed on the Media Server, terminating the ICE and DTLS sessions.

The purpose of `Location` in response header is to explicitly release resources. and if not, implicitly release resources.